File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/mediator/Developer.java
package mediator;

class Developer extends Colleague {
    public Developer(Mediator mediator) {
        super(mediator);
    }

    @Override
    public void send(String message) {
        System.out.println("Developer sending message: " + message);
        mediator.request(message, this);
    }

    @Override
    public void receive(String message) {
        System.out.println("Developer received task: " + message);
    }
}

File: code/typescript/mediator/Developer.ts
import { Colleague } from "./Colleague";
import { Mediator } from "./Mediator";

export class Developer extends Colleague {
    constructor(mediator: Mediator) {
        super(mediator);
    }

    send(message: string): void {
        console.log(`Developer sending message: ${message}`);
        this.mediator.request(message, this);
    }

    receive(message: string): void {
        console.log(`Developer received task: ${message}`);
    }
}
